warehouse/query-core/src/main/java/datawave/query/iterator/logic/EventFieldNormalizingIterator.java

```diff
@@ -0,0 +1,248 @@
+package datawave.query.iterator.logic;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import datawave.data.type.NoOpType;
+import datawave.data.type.Type;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.util.Tuple1;
+import datawave.query.util.Tuple2;
+import datawave.query.util.TypeMetadata;
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.hadoop.io.Text;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Iterator expects Event Key/Value pairs and will filter to only matching field and apply normalizations. This will result in one Key generated per normalizer
+ * applied to a field
+ */
+public class EventFieldNormalizingIterator implements SortedKeyValueIterator<Key,Value> {
```

Comment:
It seems to me that we are doing a lot of work to try and keep things sorted when all we are doing is aggregating these into a Document.  Would a better approach be to modify the aggregator to simply do the normalization?

Reply:
I originally started there, but because it has a 1:many relationship due to normalizers it was going to require interface changes to `IdentityAggregator`. We can still go with that if you prefer.

Reply:
If it makes the code simpler, then I am all for it.  We have added several new classes here whereas an api change and one simpler extension would be easier to understand and maintain.  I guess the question is how many other places will this affect my changing the interface to return 1:many relationships.

Reply:
It requires changing the parserFieldValues() to return a list rather than a single value, then updating all current methods to return singletons.